[SPEC-8103] Update GameLift server instructions for new exposed API (#869)
o3de/o3de.org: content/docs/user-guide/gems/reference/aws/aws-gamelift/gem-setup.md

                ...
                Gem::AWSGameLift.Server.Static
    )
    ```
   

> 2. **(Required)** To ensure that the Amazon GameLift Server SDK gets initialized correctly, you must mark `AWSGameLiftServerService` as required for your project server system component.

Reviewer: ```suggestion
2. **(Required)** To ensure that the Amazon GameLift Server SDK gets initialized correctly, you must indicate that the `AWSGameLiftServerService` is required for your project server system component.
```

> 2. **(Required)** In order to make sure Amazon GameLift Server SDK get initiated correctly before your project server starts, you must mark `AWSGameLiftServerService` as required for your project server system component.

Reviewer 2: "To ensure that the Amazon GameLift Server SDK gets intialized correctly...."
